### DISCUSSION TITLE
How do I import results from commercial CFD calculation software into pyMOR?

### DISCUSSION THREAD

**Discussion by liuxl1007** [pymor/pymor#2142] [Q&A]
I am a beginner and I would like to ask if it is possible to import fluent calculations? If it is possible to import them, do I still need to recreate those PDEs for fluent solving in pyMOR? Thank you very much!

**Comment by pmli** [comment]
Hi @liuxl1007, thank you for the question. There are basically two options:

1. export matrices and vectors from your software and import them into pyMOR's [`NumpyMatrixOperator`](https://docs.pymor.org/2023-1-0/autoapi/pymor/operators/numpy/index.html#pymor.operators.numpy.NumpyMatrixOperator) and [`NumpyVectorArray`](https://docs.pymor.org/2023-1-0/autoapi/pymor/vectorarrays/numpy/index.html#pymor.vectorarrays.numpy.NumpyVectorArray). Those can then be used to build full-order models in pyMOR or to directly construct data-driven reduced-order models.
2. wrap the operators and vectors in your software to use pyMOR's [`Operator`](https://docs.pymor.org/2023-1-0/autoapi/pymor/operators/interface/index.html#pymor.operators.interface.Operator) and [`VectorArray`](https://docs.pymor.org/2023-1-0/autoapi/pymor/vectorarrays/interface/index.html#pymor.vectorarrays.interface.VectorArray) interfaces. An example would be FEniCS [bindings](https://docs.pymor.org/2023-1-0/autoapi/pymor/bindings/fenics/index.html) which implement a `FenicsOperator` and `FenicsVector`. Python bindings from PyFluent might help, but I'm not sure.

Please feel free to ask more questions if you have them.
